### DISCUSSION TITLE
How to wrap few screens a custom context provider?

### DISCUSSION THREAD

**Discussion by curiosbasant** [react-navigation/react-navigation#10726] [Q&A]
https://stackoverflow.com/questions/65629534/how-to-wrap-multiple-stack-screens-with-context-provider-in-react-navigation-5

The above link is exactly what I want to do. So what's the best to way to achieve something like that?

**Comment by raajnadar** [comment]
Why not keep both providers as the parent of the `<Stack.Navigator />`?

What is the use case?  The context is available to all the children's elements, so what is the issue when wrapping the navigator?

**Reply by curiosbasant** [reply]
> 

There is no issue in doing that, but I just don't like all my screens to have a global context available unnecessary.
I just want to make it available only to some of my screens in root stack.
Would it be possible? Or do my concern evan matter?

**Reply by raajnadar** [reply]
Did you try this answer?

https://stackoverflow.com/a/70484229/5519872

**Reply by tran-simon** [reply]
the proposed answer does not work

**Comment by GunnarAK** [comment]
So did anyone find a decent solution to this? @CuriosBasant

**Reply by root-zero-o** [reply]
I just wrapped my components one by one like this. (I don't know exactly how can solve this issue. sorry.)

For example at stackoverflow, 
```javascript
<Stack.Navigator>
    <Stack.Screen name="screen1" component={Screen1} />
    <Stack.Screen name="screen2" component={Screen2} />
    <Stack.Screen name="screen3" component={Screen3} />
    <Stack.Screen name="screen4" component={Screen4} />
</Stack.Navigator/>
```

Screen1/index.tsx
```javascript
 <Provider1>
   <Screen1Component props={props} /> // props for use route and navigator
</Provider1>
```

Screen3/index.tsx
```javascript
// another provider
<Provider3>
  <Screen3Component props={props} /> // props for use route and navigator
</Provider3>
```

I seperated component for context provider(index.tsx) and component for contents(Screen1Component, Screen3Component).

If anyone got another solution, please let me know :(

**Reply by curiosbasant** [reply]
@root-zero-o But that'll create separate providers for each screen, but the whole point of the problem is to share a provider within multiple screens.

**Reply by curiosbasant** [reply]
@GunnarAK I don't think that is possible it's either whole stack or individual separate providers for each screen.
But I'm using `expo/router` now. Which serves me well. 😄  👍🏼

**Reply by root-zero-o** [reply]
I seperated Stack.Navigator for sharing provider within multiple screens.
like below, and it works.

```javascript
<Provider1>
  <Stack.Navigator>
      <Stack.Screen name="screen1" component={Screen1} />
      <Stack.Screen name="screen2" component={Screen2} />
  </Stack.Navigator/>
</Provider1>
```

```javascript
<Provider2>
 <Stack.Navigator>
      <Stack.Screen name="screen3" component={Screen3} />
      <Stack.Screen name="screen4" component={Screen4} />
  </Stack.Navigator/>
</Provider2>
```

@CuriosBasant  Anyway, I'm glad you already solved it 😄
